### PR TITLE
Feature: Demo Mode for Launcher

### PR DESCRIPTION
**Pull request by ppmathis**
This PR adds a `-d` / `--demo` boolean flag to the launcher application which can be specified to launch multiple applications consecutively without having to deploy every single time. Additionally, a new "exit" target has been added to the launcher which allows gracefully terminating the launcher if no (further) application should be launched.

A few additional changes have been added which are (more or less) related to this change:

- The bundled running configurations have been updated and renamed for consistency
- The bundled running configurations no longer launch tests by default, as these are only used by the library classes
- A new `CrowPi Run (Demo Mode)` running configuration has been added
- Two flaky tests related to HumiTemp and IrReceiver have been fixed to avoid failures
- The `CrowPiPlatform` now contains a helper method for building a Pi4J context to slim down the launcher code
- The `ButtonMatrixApp` and `LedMatrixApp` have been slightly adjusted to properly free up their resources
- The `SevenSegmentApp` no longer runs infinitely and instead exits after 15 seconds

This task is tracked as ppmathis/fhnw-crowpi#58 and a PR which updates the relevant documentation will follow soon™.

### CODE REVIEW

File: .run/crowpi-debug.run.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <component name="ProjectRunConfigurationManager">
-  <configuration default="false" name="crowpi-examples [debug]" type="MavenRunConfiguration" factoryName="Maven">
+  <configuration default="false" name="CrowPi Debug" type="MavenRunConfiguration" factoryName="Maven">
```
Comment (tobiassiegrist) [inline .run/crowpi-debug.run.xml:2]: Do you know that if you change this name you need to update screenshots and documentation as well?

File: .run/crowpi-remote-debug.run.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <component name="ProjectRunConfigurationManager">
-  <configuration default="false" name="Remote Debug" type="Remote">
+  <configuration default="false" name="CrowPi Remote Debug" type="Remote">
```
Comment (tobiassiegrist) [inline .run/crowpi-remote-debug.run.xml:2]: Same as above

File: .run/crowpi-run.run.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <component name="ProjectRunConfigurationManager">
-  <configuration default="false" name="crowpi-examples [install]" type="MavenRunConfiguration" factoryName="Maven">
+  <configuration default="false" name="CrowPi Run" type="MavenRunConfiguration" factoryName="Maven">
```
Comment (tobiassiegrist) [inline .run/crowpi-run.run.xml:2]: same as above

**Review by tobiassiegrist — CHANGES_REQUESTED** [review]
Nice job. Makes it quite simple to just enjoy all the CrowPi features. Just a small point we need to discuss / you need to be aware of. 

Demo mode LGTM 😃

File: .run/crowpi-debug.run.xml (hunk 1 above)
Comment (ppmathis) [reply .run/crowpi-debug.run.xml:2]: I'm well aware of that and noticed that due to now skipping tests by default and a couple other changes I have to touch these parts of the documentation anyways, so I will quickly update the screenshots too. This is already a part of the updated documentation PR I'm working on.

**Review by tobiassiegrist — APPROVED** [review]

**Pull request merged**